www/class/config-generate/host.class.php

```diff
@@ -585,4 +588,29 @@ public function reset()
         $this->generatedHosts = array();
         parent::reset();
     }
+
+    /**
+     * @param array<string,mixed> $host
+     * @return self
+     */
+    private function addHostToHostCategoryMembers(array &$host): self
+    {
+        if (!isset($host['hc'])) {
+            $stmt = $this->backend_instance->db->prepare(
+                "SELECT hostcategories_hc_id
+                FROM hostcategories_relation
+                WHERE host_host_id = :host_id"
+            );
+            $stmt->bindParam(':host_id', $host['host_id'], PDO::PARAM_INT);
+            $stmt->execute();
+            $host['hc'] = $stmt->fetchAll(PDO::FETCH_COLUMN);
+        }
```

Comment:
This function is doing more than one thing : retrieves host categories from database if not set, injects them in argument passed by reference and updates ''global'' HostCategory instance.

Are you sure about passing of $host argument by reference ?
Maybe you should separate host category fetching from host category add. In this case you would pass $host and $hostCategory as arguments to this function ?

Reply:
This function has been moved to the HostAbstract class, refactored and renamed as "insertHostInHostCategoryMembers" but still use argument passed by reference as it is based on legacy code